src/main/java/org/onedatashare/server/module/vfs/VfsResource.java

```diff
@@ -220,7 +220,7 @@ public Long buildDirectoryTree(List<Stat> sub, FileObject[] fileObjects, String
                             }
                             sink.next(new Slice(b));
                         } else {
-                            int remaining = sizeInt - state;
+                            int remaining = Math.toIntExact(sizeInt - state);
```

Comment:
Consider a file of size 2 TB (hypothetical situation for now -> to consider a situation that exceeds the 'int' range)
sizeInt = 2*10^12
state = 0 (first run state will be 0)

Won't this line of code throw an exception? This was what I was trying tell in the lab.

Reply:
I explained to you during the meeting, This can not happen because of the if statement.
State can not be 0 in this case because sizeInt-state < slicesize

Reply:
Please read the if statement again and think about what range of number state can be!

Reply:
ok